### PR TITLE
Raneet10/handler test

### PR DESCRIPTION
**Pull request by Raneet10**

### CODE REVIEW

File: x/pooltoy/handler_test.go [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+	require.NoError(t, err1)
+
+	//Test for Creator who isn't an admin
+	TestUserAddress2 := GenerateTestAddress(3)
+	testMsg2 := types.NewMsgCreateUser(TestCreatorAddress, TestUserAddress2, false, "test_two", "test2@test.com")
+	creator := keeper.GetUserByAccAddress(ctx, TestCreatorAddress)
+	_, err2 := handler(ctx, testMsg2)
+	testErr1 := sdkerrors.Wrap(sdkerrors.ErrUnauthorized, fmt.Sprintf("user %s (%s) is not an admin", creator.Name, TestCreatorAddress))
+	require.Error(t, err2)
+	require.True(t, errors.Is(err2, testErr1))
```
Comment (okwme) [inline x/pooltoy/handler_test.go:43]:
```suggestion
	_, err2 := handler(ctx, testMsg2)
	require.Error(t, err2)

	creator := keeper.GetUserByAccAddress(ctx, TestCreatorAddress)
	testErr1 := sdkerrors.Wrap(sdkerrors.ErrUnauthorized, fmt.Sprintf("user %s (%s) is not an admin", creator.Name, TestCreatorAddress))
	require.True(t, errors.Is(err2, testErr1))
```

File: x/pooltoy/handler_test.go [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+	//Test for non-existent Creator
+	testMsg3 := types.NewMsgCreateUser(TestCreatorAddress, TestUserAddress1, true, "test_three", "test3@test.com")
+	_, err3 := handler(ctx, testMsg3)
+	testErr2 := sdkerrors.Wrap(sdkerrors.ErrUnauthorized, fmt.Sprintf("user %s does not exist", TestCreatorAddress))
+	require.Error(t, err3)
+	require.True(t, errors.Is(err3, testErr2))
```
Comment (okwme) [inline x/pooltoy/handler_test.go:50]:
```suggestion
	_, err3 := handler(ctx, testMsg3)
	require.Error(t, err3)

	testErr2 := sdkerrors.Wrap(sdkerrors.ErrUnauthorized, fmt.Sprintf("user %s does not exist", TestCreatorAddress))
	require.True(t, errors.Is(err3, testErr2))
```

File: x/pooltoy/handler_test.go [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+	//Test for duplicate user TestUserAddress1 where creator is the user itself
+	testMsg4 := types.NewMsgCreateUser(TestUserAddress1, TestUserAddress1, true, "test_one", "test1@test.com")
+	_, err4 := handler(ctx, testMsg4)
+	testErr3 := sdkerrors.Wrap(sdkerrors.ErrInvalidAddress, fmt.Sprintf("user %s already exists", TestUserAddress1))
+	require.Error(t, err4)
+	require.True(t, errors.Is(err4, testErr3))
```
Comment (okwme) [inline x/pooltoy/handler_test.go:57]:
```suggestion
	_, err4 := handler(ctx, testMsg4)
	require.Error(t, err4)

	testErr3 := sdkerrors.Wrap(sdkerrors.ErrInvalidAddress, fmt.Sprintf("user %s already exists", TestUserAddress1))
	require.True(t, errors.Is(err4, testErr3))
```

**Review by okwme — APPROVED** [review]
thanks so much for writing this test! Sorry the changes I suggested are just stylistic and really just a preference for organizing. If you find them helpful feel free to accept them but I'd be happy to merge your changes without them as well!

**Comment by Raneet10** [comment]
Thanks for making the changes! Of course, it's better to have clarity.

**Pull request merged**